analyzers/src/SonarAnalyzer.Common/Helpers/SonarLintXmlReader.cs

```diff
@@ -0,0 +1,93 @@
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using System.IO;
+using System.Text;
+using System.Xml;
+using System.Xml.Serialization;
+using Microsoft.CodeAnalysis.Text;
+
+namespace SonarAnalyzer.Helpers;
+
+public class SonarLintXmlReader
+{
+    public static readonly SonarLintXmlReader Empty = new(null, LanguageNames.CSharp);
+
+    private readonly SonarLintXml sonarLintXml;
+    private readonly string propertyLanguage;
+
+    private bool? ignoreHeaderComments;
+    public bool IgnoreHeaderComments => ignoreHeaderComments ??= ReadBoolean(ReadProperty($"sonar.{propertyLanguage}.ignoreHeaderComments"));
+
+    private bool? analyzeGeneratedCode;
+    public bool AnalyzeGeneratedCode => analyzeGeneratedCode ??= ReadBoolean(ReadProperty($"sonar.{propertyLanguage}.analyzeGeneratedCode"));
+
+    private string[] exclusions;
+    public string[] Exclusions => exclusions ??= ReadCommaSeparatedArray(ReadProperty("sonar.exclusions"));
+
+    private string[] inclusions;
+    public string[] Inclusions => inclusions ??= ReadCommaSeparatedArray(ReadProperty("sonar.inclusions"));
+
+    private string[] globalExclusions;
+    public string[] GlobalExclusions => globalExclusions ??= ReadCommaSeparatedArray(ReadProperty("sonar.global.exclusions"));
+
+    private string[] testExclusions;
+    public string[] TestExclusions => testExclusions ??= ReadCommaSeparatedArray(ReadProperty("sonar.test.exclusions"));
+
+    private string[] testInclusions;
+    public string[] TestInclusions => testInclusions ??= ReadCommaSeparatedArray(ReadProperty("sonar.test.inclusions"));
+
+    private string[] globalTestExclusions;
+    public string[] GlobalTestExclusions => globalTestExclusions ??= ReadCommaSeparatedArray(ReadProperty("sonar.global.test.exclusions"));
+
+    public SonarLintXmlReader(SourceText sonarLintXml, string language)
+    {
+        this.sonarLintXml = sonarLintXml == null ? SonarLintXml.Empty : ParseContent(sonarLintXml);
+        propertyLanguage = language == LanguageNames.CSharp ? "cs" : "vbnet";
+    }
+
+    private static SonarLintXml ParseContent(SourceText sonarLintXml)
+    {
+        try
+        {
+            var serializer = new XmlSerializer(typeof(SonarLintXml));
+            var byteArray = Encoding.UTF8.GetBytes(sonarLintXml.ToString());
+            var stream = new MemoryStream(byteArray);
+            using var sr = new StreamReader(stream, Encoding.UTF8, false);
+            using var reader = XmlReader.Create(sr);
+            return (SonarLintXml)serializer.Deserialize(reader);
```

Comment:
I had a bunch of comments here and deleted them (unreleased memory stream, too many useless classes). Delete the method, take a look at `ProjectConfigReader` constructor and `ReadContent` method.  And do here exactly the same thing as it's there.

Including the action to read the content directly in constructor as mentioned above

Reply:
The initial implementation was exactly that, but then Martin raised [a comment](https://github.com/SonarSource/sonar-dotnet/pull/6828#discussion_r1120490770) about a potential under-the-hood conversion from UTF-8 (most XML files are UTF-8 encoded) to UTF-16.

Reply:
That's too much handling for a problem that we don't have.

Reply:
@cristian-ambrosini-sonarsource how was this comment handled?

Reply:
Ok, I'll revert to the previous implementation here #6939.